Mathlib/Algebra/BigOperators/Pi.lean

```diff
@@ -181,3 +181,11 @@ def Pi.monoidHomMulEquiv {ι : Type*} [Fintype ι] [DecidableEq ι] (M : ι →
       MonoidHom.mul_apply, mul_apply]
 
 end MulEquiv
+
+@[to_additive (attr := elab_as_elim)]
+lemma Pi.induction_mul {ι M : Type*} [Finite ι] [DecidableEq ι] [CommMonoid M] (p : (ι → M) → Prop)
+    (f : ι → M) (hone : p 1) (hmul : ∀ f g, p f → p g → p (f * g))
+    (hsingle : ∀ i m, p (Pi.mulSingle i m)) : p f := by
```

Comment:
Convention for indunction principles is to drop the `h`s:
```suggestion
    (f : ι → M) (one : p 1) (mul : ∀ f g, p f → p g → p (f * g))
    (mulSingle : ∀ i m, p (Pi.mulSingle i m)) : p f := by
```
You might want to manually `to_additive` this so that the names are `add` and `single` for the additive version.